include/pstl/internal/algorithm_impl.h

```diff
@@ -133,186 +138,197 @@ void pattern_walk_brick( Iterator first, Iterator last, Brick brick, /*parallel=
 //------------------------------------------------------------------------
-template<class Iterator, class Function>
-void brick_it_walk1( Iterator first, Iterator last, Function f, /*vector=*/std::false_type ) noexcept {
-    for(; first!=last; ++first )
-        f(first);
+template<class _ForwardIterator, class _Function>
+void brick_it_walk1( _ForwardIterator __first, _ForwardIterator __last, _Function __f, /*vector=*/std::false_type ) noexcept {
+    for(; __first != __last; ++__first )
+        __f(__first);
 }
 
-template<class Iterator, class Function>
-void brick_it_walk1( Iterator first, Iterator last, Function f, /*vector=*/std::true_type ) noexcept {
-    unseq_backend::simd_it_walk_1(first, last-first, f);
+template<class _RandomAccessIterator, class _Function>
+void brick_it_walk1( _RandomAccessIterator __first, _RandomAccessIterator __last, _Function __f, /*vector=*/std::true_type ) noexcept {
+    unseq_backend::simd_it_walk_1(__first, __last - __first, __f);
 }
 
-template<class Iterator, class Function, class IsVector>
-void pattern_it_walk1( Iterator first, Iterator last, Function f, IsVector is_vector, /*parallel=*/std::false_type ) noexcept {
-    brick_it_walk1( first, last, f, is_vector );
+template<class __ForwardIterator, class _Function, class _IsVector>
+void pattern_it_walk1( __ForwardIterator __first, __ForwardIterator __last, _Function __f, _IsVector __is_vector,
+                       /*parallel=*/std::false_type ) noexcept {
+    internal::brick_it_walk1( __first, __last, __f, __is_vector );
 }
 
-template<class Iterator, class Function, class IsVector>
-void pattern_it_walk1( Iterator first, Iterator last, Function f, IsVector is_vector, /*parallel=*/std::true_type ) {
-    except_handler([=]() {
-        par_backend::parallel_for( first, last, [f,is_vector](Iterator i, Iterator j) {
-            brick_it_walk1(i,j,f,is_vector);
+template<class __ForwardIterator, class _Function, class _IsVector>
+void pattern_it_walk1( __ForwardIterator __first, __ForwardIterator __last, _Function __f, _IsVector __is_vector,
+                       /*parallel=*/std::true_type ) {
+    internal::except_handler([=]() {
+        par_backend::parallel_for( __first, __last, [__f,__is_vector](__ForwardIterator __i, __ForwardIterator __j) {
+            internal::brick_it_walk1( __i, __j, __f, __is_vector );
         });
     });
 }
 
 //------------------------------------------------------------------------
 // walk1_n
 //------------------------------------------------------------------------
-template<class InputIterator, class Size, class Function>
-InputIterator brick_walk1_n(InputIterator first, Size n, Function f, /*IsVectorTag=*/std::false_type ) {
-    return internal::for_each_n( first, n, f ); // calling serial version
+template<class InputIterator, class _Size, class _Function>
+InputIterator brick_walk1_n(InputIterator __first, _Size __n, _Function __f, /*_IsVectorTag=*/std::false_type ) {
```

Comment:
Missing underscores before InputIterator and RandomAccessIterator until brick_walk2